tests/integration/test_opensearch.py

```diff
@@ -0,0 +1,214 @@
+
+    tls_config = {"generate-self-signed-certificates": "true", "ca-common-name": "CN_CA"}
+    # Set kernel params in model config opensearch can run
+    model_config = {
+        "logging-config": "<root>=INFO;unit=DEBUG",
+        "update-status-hook-interval": "1m",
+        "cloudinit-userdata": """postruncmd:
+            - [ 'sysctl', '-w', 'vm.max_map_count=262144' ]
+            - [ 'sysctl', '-w', 'fs.file-max=1048576' ]
+            - [ 'sysctl', '-w', 'vm.swappiness=0' ]
+            - [ 'sysctl', '-w', 'net.ipv4.tcp_retries2=5' ]
+        """,
+    }
+    await ops_test.model.set_config(model_config)
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            OPENSEARCH[ops_test.cloud_name],
+            channel="edge",
+            application_name=OPENSEARCH[ops_test.cloud_name],
+            num_units=3,
+        ),
+        ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="edge", config=tls_config),
+        ops_test.model.deploy(
+            data_integrator_charm, application_name="data-integrator", num_units=1, series="jammy"
+        ),
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[DATA_INTEGRATOR, OPENSEARCH[ops_test.cloud_name], TLS_CERTIFICATES_APP_NAME],
+        idle_period=10,
+        timeout=1600,
+    )
+    config = {"index-name": INDEX_NAME, "extra-user-roles": OPENSEARCH_EXTRA_USER_ROLES}
+    await ops_test.model.applications[DATA_INTEGRATOR].set_config(config),
+    await asyncio.gather(
+        ops_test.model.relate(OPENSEARCH[ops_test.cloud_name], TLS_CERTIFICATES_APP_NAME),
+        ops_test.model.relate(DATA_INTEGRATOR, OPENSEARCH[ops_test.cloud_name]),
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[DATA_INTEGRATOR, OPENSEARCH[ops_test.cloud_name], TLS_CERTIFICATES_APP_NAME],
+        status="active",
+        idle_period=10,
+        timeout=1600,
+    )
+
+
+async def test_sending_requests_using_opensearch(ops_test: OpsTest):
+    """Verifies intended use case of data-integrator charm.
+
+    This test verifies that we can use the credentials provided to the data-integrator charm to
+    update and retrieve data from the opensearch charm.
+    """
+    if ops_test.cloud_name != "localhost":
+        pytest.skip("opensearch does not have a k8s charm yet.")
+
+    await ops_test.model.wait_for_idle(
+        apps=[DATA_INTEGRATOR, OPENSEARCH[ops_test.cloud_name], TLS_CERTIFICATES_APP_NAME],
+        status="active",
+        idle_period=30,
+        timeout=1000,
+    )
+
+    # get credentials for opensearch
+    credentials = await fetch_action_get_credentials(
+        ops_test.model.applications[DATA_INTEGRATOR].units[0]
+    )
+
+    album_payload = (
+        '{"artist": "Vulfpeck", "genre": ["Funk", "Jazz"], "title": "Thrill of the Arts"}'
+    )
+    # This request can be temperamental, because opensearch can appear active without having
+    # available databases.
+    opensearch_request(
```

Comment:
Is it possible to move the interaction with Opensearch in the testing [app](https://github.com/canonical/data-integrator/tree/main/tests/integration/app-charm)? All other products are tested in this way to have a single testing app that works both with VM and K8S charms.

Reply:
It is, but deploying a whole app to communicate with opensearch is unnecessary if we have the correct credentials. I feel this test process tests the intended usage of the data-integrator charm, but I'm happy to add this into the test charm if you want.

Reply:
I prefer to have it in the app. This will keep all tests uniform and it will also work for the opensearch k8s operator

Reply:
Ok, I'll add that in :)